doc/contributing.rst

```diff
@@ -1,221 +1,760 @@
+command:
+
+.. code-block:: console
+
+    $ # see what state the local copy of the codebase is in
+    $ git status
+
+Those unstaged changes can be `added`_ to the stage one by one, by either
+adding a whole file's worth of changes, or by adding only certain lines
+interactively:
+
+.. code-block:: console
+
+    $ # add a whole file's worth of changes
+    $ # (same command works to add a completely new file):
+    $ git add mne/some_file.py
+    $ # enter interactive staging mode, to add only portions of the file:
+    $ git add -p mne/viz/some_other_file.py
+
+Once you've collected all the related changes together on the stage, the ``git
+status`` command will now refer to them as "changes staged for commit". You can
+commit them to the current branch with the `commit`_ command. If you just type
+``git commit`` by itself, git will open the text editor you configured it to
+use so that you can write a *commit message* — a short description of the
+changes you've grouped together in this commit. You can bypass the text editor
+by passing a commit message on the command line with the ``-m`` flag. For
+example, if your first commit adds a new feature, your commit message might be:
+
+.. code-block:: console
+
+    $ git commit -m 'ENH: adds feature X to the Epochs class'
+
+Once you've made the commit, the stage is now empty, and you can repeat the
+cycle, adding the unit tests and documentation changes:
+
+.. code-block:: console
+
+    $ git add mne/tests/some_testing_file.py
+    $ git commit -m 'add test of new feature X of the Epochs class'
+    $ git add -p mne/some_file.py mne/viz/some_other_file.py
+    $ git commit -m 'DOC: update Epochs and BaseEpochs docstrings'
+    $ git add tutorials/new_tutorial_file.py
+    $ git commit -m 'DOC: adds new tutorial about feature X'
+
+When you're done, it's time to run the test suite to make sure your changes
+haven't broken any existing functionality, and to make sure your new test
+covers the lines of code you've added (see :ref:`run-tests` and
+:ref:`build-docs`, below). Once everything looks good, it's time to push your
+changes to your fork:
+
+.. code-block:: console
+
+    $ git push origin new-feature-x
+
+Finally, go to the `MNE-Python GitHub`_ page, click on the pull requests tab,
+click the "new pull request" button, and choose "compare across forks" to
+select your new branch (``new-feature-x``) as the "head repository".  See the
+GitHub help page on `creating a PR from a fork`_ for more information about
+opening pull requests.
+
+If any of the tests failed before you pushed your changes, try to fix them,
+then add and commit the changes that fixed the tests, and push to your fork. If
+you're stuck and can't figure out how to fix the tests, go ahead and push your
+commits to your fork anyway and open a pull request (as described above), then
+in the pull request you should describe how the tests are failing and ask for
+advice about how to fix them.
+
```

Comment:
Should one add a short instruction about when and how to rebase?
And `git branch -d/D new-feature-x`, resp. `git push origin -d new-feature-x` and `git prune origin`, to complete the PR work cycle?

Reply:
Let's add this later. It should maybe be its own page